var/spack/repos/builtin/packages/py-amrex/package.py

```diff
@@ -113,7 +113,10 @@ def copy_test_sources(self):
         install test subdirectory for use during `spack test run`."""
         cache_extra_test_sources(self, [self.tests_src_dir])
 
-    def test(self):
+    def test_pytest(self):
         """Perform smoke tests on the installed package."""
-        pytest = which("pytest")
-        pytest(join_path(install_test_root(self), self.tests_src_dir))
+        test_dir = join_path(self.test_suite.current_test_cache_dir, self.tests_src_dir)
+        with working_dir(test_dir):
+            pytest = which("pytest")
+            assert pytest is not None, "Cannot find pytest"
+            pytest(join_path(install_test_root(self), self.tests_src_dir))
```

Comment:
This is still trying to run under the install prefix.

```suggestion
            pytest()
```